src/daft-logical-plan/src/builder/resolve_expr.rs

```diff
@@ -151,6 +151,80 @@ fn col_resolves_to_plan(column: &UnresolvedColumn, plan: &LogicalPlanRef) -> Daf
     })
 }
 
+fn replace_element_with_column_ref(expr: ExprRef, replacement: ExprRef) -> DaftResult<ExprRef> {
+    expr.transform(|e| {
+        if matches!(
+            e.as_ref(),
+            Expr::Column(Column::Unresolved(UnresolvedColumn {
+                name,
+                plan_ref: PlanRef::Unqualified,
+                plan_schema: None
+            })) if name.as_ref() == ""
+        ) {
+            Ok(Transformed::yes(replacement.clone()))
+        } else {
+            Ok(Transformed::no(e))
+        }
+    })
+    .map(|res| res.data)
+}
+
+fn resolve_list_evals(expr: ExprRef) -> DaftResult<ExprRef> {
+    if expr.has_element() {
+        expr.transform_down(|e| {
```

Comment:
No need for the `expr.has_element()` I think. Since it has to traverse the entire expression tree anyway, it does not improve performance. We can just run this on every expression.